### PR TITLE
[LPT] Fix concat transformation tests and add subtract_convert push support

### PR DESCRIPTION
**Pull request by ndemashov**

### CODE REVIEW

File: src/common/low_precision_transformations/src/concat.cpp [hunk 1]
```diff
@@ -111,9 +117,21 @@ bool ConcatTransformation::transform(TransformationContext& context, ngraph::pat
 
         if (!allDequantizationShiftAreZero) {
             auto subtractInput = dequantization.subtract == nullptr ?
-                std::make_shared<ngraph::opset1::Constant>(deqPrecision, targetShape, std::vector<float>({ 0.f })) :
+                (allDequantizationShiftConvertAreNotZero ?
+                    std::make_shared<ngraph::opset1::Constant>(
+                        element::u8,
+                        targetShape,
+                        std::vector<unsigned int>({ 0u })) :
```
Comment (eshoguli) [inline src/common/low_precision_transformations/src/concat.cpp:124]: 1. `element::u8`: use original precision from the model, please.
2. minor: `std::vector<unsigned int>({ 0u }))`: use more appropriate general type instead `unsigned int`. I think `float` should be used here.

File: src/common/low_precision_transformations/src/concat.cpp [hunk 2]
```diff
@@ -55,17 +55,22 @@ bool ConcatTransformation::transform(TransformationContext& context, ngraph::pat
...
         if (dequantization.multiply != nullptr) {
             allDequantizationMultiplyAreZero = false;
         }
 
-        if (!allDequantizationShiftAreZero && !allDequantizationMultiplyAreZero) {
+        if (!allDequantizationShiftAreZero && !allDequantizationMultiplyAreZero &&
+            !allDequantizationShiftConvertAreNotZero) {
             break;
         }
     }
```
Comment (eshoguli) [inline src/common/low_precision_transformations/src/concat.cpp:76]: `allDequantizationShiftConvertAreNotZero` variable initialization is not correct. If there are no `Substract` operations  then `allDequantizationShiftConvertAreNotZero` will be `true`.

Comment (ndemashov) [reply src/common/low_precision_transformations/src/concat.cpp:79]: Now if allDequantizationShiftAreZero is true after loop then set allDequantizationShiftConvertAreNotZero false. Thank you

File: src/common/low_precision_transformations/src/concat.cpp (hunk 1 above)
Comment (ndemashov) [reply src/common/low_precision_transformations/src/concat.cpp]: PrecisionBeforeConvert is now used to save the original precision of the model. Thank you

**Review by eshoguli — APPROVED** [review]

**Pull request merged**